internal/resources/cluster_onoff_schedule.go

```diff
@@ -0,0 +1,392 @@
+}
+
+// Metadata returns the OnOffSchedule resource type name.
+func (c *ClusterOnOffSchedule) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_onoff_schedule"
+
+}
+
+// Schema defines the schema for the OnOffSchedule resource.
+func (c *ClusterOnOffSchedule) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = OnOffScheduleSchema()
+}
+
+// Create creates a new OnOffSchedule.
+func (c *ClusterOnOffSchedule) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.ClusterOnOffSchedule
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := c.validateCreateClusterOnOffScheduleRequest(plan)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error parsing create cluster on/off schedule request",
+			"Could not create cluster on/off schedule "+err.Error(),
+		)
+		return
+	}
+
+	var organizationId = plan.OrganizationId.ValueString()
+	var projectId = plan.ProjectId.ValueString()
+	var clusterId = plan.ClusterId.ValueString()
+
+	var days = make([]scheduleapi.DayItem, 0)
+	for _, d := range plan.Days {
+		days = append(days, scheduleapi.DayItem{
+			State: d.State.ValueString(),
+			Day:   d.Day.ValueString(),
+			From: &scheduleapi.OnTimeBoundary{
+				Hour:   d.From.Hour.ValueInt64(),
+				Minute: d.From.Minute.ValueInt64(),
+			},
+			To: &scheduleapi.OnTimeBoundary{
+				Hour:   d.To.Hour.ValueInt64(),
+				Minute: d.To.Minute.ValueInt64(),
+			},
+		})
+	}
+
+	scheduleRequest := scheduleapi.CreateClusterOnOffScheduleRequest{
+		Timezone: plan.Timezone.ValueString(),
+		Days:     days,
+	}
+
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/onOffSchedule", c.HostURL, organizationId, projectId, clusterId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPost, SuccessStatus: http.StatusNoContent}
+	_, err = c.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		scheduleRequest,
+		c.Token,
+		nil,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error executing request",
+			errorMessageWhileOnOffScheduleCreation+api.ParseError(err),
+		)
+		return
+	}
+
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	refreshedState, err := c.retrieveClusterOnOffSchedule(ctx, organizationId, projectId, clusterId)
+	if err != nil {
+		resp.Diagnostics.AddWarning(
+			"Error Reading Capella Cluster On/Off Schedule",
+			"Could not read Capella Cluster On/Off Schedule for the cluster: %s "+clusterId+"."+errorMessageAfterBackupScheduleCreation+api.ParseError(err),
+		)
+		return
+	}
+
+	// Set state to fully populated data
+	diags = resp.State.Set(ctx, refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+}
+
+// Read reads OnOffSchedule information.
```

Comment:
these godocs aren't informative, either we can remove them or make them detailed.